### PR TITLE
JS error fixes and updates to track an ebook download

### PR DESCRIPTION
**Pull request by gib**

### CODE REVIEW

**Comment by gib** [comment]
Not my best, but the segment traffic looks good to me (logs the book). I'm going to merge and make sure staging is happy, but do you mind reviewing @sandersonet?

File: src/assets/javascript/newsletter.coffee [hunk 1]
```diff
@@ -27,25 +27,43 @@ $ ->
...
               traits =
-                email: evamilVal,
+                email: emailVal,
                 newsletter_subscribed: true,
                 name: "#{fname.val()} #{lname.val()}",
                 company_name: company.val()
               analytics.identify emailVal, traits
 
-            analytics.track 'Subscribed To Newsletter', { email: emailVal }
+            analytics.track 'Subscribed To Newsletter', traits
```
Comment (sandersonet) [inline src/assets/javascript/newsletter.coffee]: I think we should remove this event

**Comment by sandersonet** [comment]
I think it might help to refactor this a bit to remove all of the old newsletter functionality.

**Comment by gib** [comment]
Can do... I thought it would still be useful as a way to find people that have essentially said we can email them? I have much to learn in the newsletter sign up areas of common courtesies.

**Comment by gib** [comment]
Thanks @sandersonet, newsletter out, ebook in.

**Pull request merged**